.github/workflows/maven.yml

```diff
@@ -33,4 +33,18 @@ jobs:
       run: mvn -B package -DskipTests --file pom.xml
     - name: Test with Maven
       run: mvn -B verify --file pom.xml
-    - uses: codecov/codecov-action@v2
+    - name: Add coverage to PR
+      id: jacoco
+      uses: madrapps/jacoco-report@v1.6.1
+      with:
+        paths: ${{ github.workspace }}/**/target/site/jacoco-merged-test-coverage-report/jacoco.xml
+        token: ${{ secrets.GITHUB_TOKEN }}
+        min-coverage-overall: 80
+        min-coverage-changed-files: 80
+        title: Coverage Report # A title is required for update-comment to work correctly
+        update-comment: true
+        pass-emoji: ':white_check_mark:'
+        fail-emoji: ':x:'
+        continue-on-error: true # Defaults to true. If false, fails check if coverage below minimum threshold
```

Comment:
I would either remove these three lines of comments completely, or just leave this: `# The repository is currently far below the threshold so we can't enforce it for now`. The reason for this is because the information about the default values is not important as we're overwriting it (and it's in the documentation anyway), and the last sentence is more of a comment that belongs in a PR or in a discussion somewhere else.